### PR TITLE
Add cmdLineTester_hangTest to OpenJ9

### PR DESCRIPTION
**Pull request by OscarQQ**
related: https://github.com/eclipse-openj9/openj9/issues/12851
[ci skip]

Signed-off-by: Yixin Qian <Yixin.Qian@ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins line endings check

**Comment by pshipton** [comment]
@OscarQQ the line endings check is failing
```
16:29:35  test/functional/cmdLineTests/hangTest/build.xml:79: trailing whitespace.
16:29:35  +			</and>	
16:29:35  test/functional/cmdLineTests/hangTest/hang.xml:37: space before tab in indent.
16:29:35  + 	</test>
16:29:35  test/functional/cmdLineTests/hangTest/src/org/openj9/test/hangTest/Hang.java:32: trailing whitespace.
16:29:35  +    
```

**Comment by llxia** [comment]
Oscar's internship has ended, but he is willing to help with PR. Thanks Oscar!

File: test/functional/cmdLineTests/hangTest/playlist.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  [2] http://openjdk.java.net/legal/assembly-exception.html
+
+  SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+<playlist xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../../TKG/playlist.xsd">
+	<test>
+		<testCaseName>cmdLineTester_hangTest</testCaseName>
+		<disable>
+			<platform>.*windows.*</platform>
+		</disable>
```
Comment (llxia) [inline test/functional/cmdLineTests/hangTest/playlist.xml:28]: The disable syntax has changed. Could you update it?

```
  <disables>
    <disable>
      <comment>reason or issue for exclude</comment>
      <platform>.*windows.*</platform>
    </disable>
  </disables>
```

File: test/functional/cmdLineTests/hangTest/playlist.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		<variations>
+			<variation>NoOptions</variation>
+		</variations>
+		<command>$(ADD_JVM_LIB_DIR_TO_LIBPATH) $(JAVA_COMMAND) $(JVM_OPTIONS) -Xdump \
```
Comment (llxia) [inline test/functional/cmdLineTests/hangTest/playlist.xml:32]: Do we need ADD_JVM_LIB_DIR_TO_LIBPATH and -Xdump?

File: test/functional/cmdLineTests/hangTest/build.xml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	<target name="build" >
+		<if>
+			<and>
+				<or>
+					<equals arg1="${JDK_IMPL}" arg2="ibm"  />
+					<equals arg1="${JDK_IMPL}" arg2="openj9" />
+				</or>
+				<not>
+					<equals arg1="${JDK_VERSION}" arg2="8" />
+				</not>
```
Comment (llxia) [inline test/functional/cmdLineTests/hangTest/build.xml:78]: This test should work for all versions

File: test/functional/cmdLineTests/hangTest/playlist.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	-DJARPATH=$(Q)$(TEST_RESROOT)$(D)hangTest.jar$(Q) \
+	-DEXE=$(SQ)bash $(TEST_RESROOT)$(D)hang.sh $(JAVA_COMMAND) $(TEST_RESROOT)$(D)hangTest.jar $(SQ) -jar $(CMDLINETESTER_JAR) \
```
Comment (llxia) [inline test/functional/cmdLineTests/hangTest/playlist.xml:34]: $(JVM_OPTIONS) needs to be passed in. i.e., `-DEXE=$(SQ)bash $(TEST_RESROOT)$(D)hang.sh $(JAVA_COMMAND) $(JVM_OPTIONS) $(TEST_RESROOT)$(D)hangTest.jar $(SQ) -jar $(CMDLINETESTER_JAR)`

**Comment by llxia** [comment]
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/252/

File: test/functional/cmdLineTests/hangTest/playlist.xml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+			<variation>NoOptions</variation>
+		</variations>
+		<command> $(JAVA_COMMAND) $(JVM_OPTIONS) \
+	-DJARPATH=$(Q)$(TEST_RESROOT)$(D)hangTest.jar$(Q) \
+	-DEXE=$(SQ)bash $(TEST_RESROOT)$(D)hang.sh $(JAVA_COMMAND) $(JVM_OPTIONS) $(TEST_RESROOT)$(D)hangTest.jar $(SQ) -jar $(CMDLINETESTER_JAR) \
+	-config $(Q)$(TEST_RESROOT)$(D)hang.xml$(Q) \
+	-xids all,$(PLATFORM) -explainExcludes -nonZeroExitWhenError; \
+	$(TEST_STATUS)</command>
+		<levels>
+			<level>sanity</level>
```
Comment (llxia) [inline test/functional/cmdLineTests/hangTest/playlist.xml:42]: One more thing, could we set this in extended?

**Comment by llxia** [comment]
jenkins test extended all jdk8,jdk11

**Review by llxia — APPROVED** [review]
Thanks @OscarQQ !

**Pull request merged**